### PR TITLE
fix(loading-regions): state loading stuck when "US" is not in the supported countries

### LINKED ISSUE #3059: [bug]: Shipping Information Form Stuck at 'Loading Regions...'
**Describe the bug**

Guest Shipping Information Form Stuck at 'Loading Regions...'

**To reproduce**

1. Make a store with only one country available and the same default country. (Not "United States")
2. Try to enter your address in guest checkout.
3. Scroll down 'Region' and try to select a region.
4. The select/dropdown is stuck at 'Loading Regions'.

**Expected behavior**

You should be able to select a region.

**Screenshots**

Screenshot 2021-03-15 at 16 12 07

**Possible solutions**

N/A

**Debug Report**

N/A

**Please complete the following device information:**

-   Device: Mac
-   Browser: Chrome and Safari
-   Browser Version: Latest
-   Magento Version: 2.4.1

**Please let us know what packages this bug is in regards to:**

-   [ ] `venia-concept`
-   [x] `venia-ui`
-   [ ] `pwa-buildpack`
-   [x] `peregrine`
-   [ ] `pwa-devdocs`
-   [ ] `upward-js`
-   [ ] `upward-spec`
-   [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by HaidarZ**
## Description

I've added an environment variable that fallbacks to **"US"** and replace all occurence of the latter country by `process.env.DEFAULT_COUNTRY_CODE`.  
This patch will allow the solution developer to specify the default selected country as he likes. If it's not specified it will fallback to **"US"**.  
Please note that  **Default Country** is an option in the magento store configuration but unfortunately it's not return neither bu the graphql nor by the rest api. 

## Related Issue

Closes #3059.

### Verification Steps

1. Go to the .env file in venia-concept and uncomment `DEFAULT_COUNTRY_CODE`
2. Write down your preferred default country
3. Add anything to the cart and go to checkout page while guest to see your preferred country selected by default.

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by sirugh — DISMISSED** [review]
A few requests, but this seems to work. Thanks for updating the translations as well. 

I do think the preferable solution will be to use a queryable value at runtime so that we can manage more than one store view with different default countries, but that requires some support from the backend, which means waiting for the next release. In the interim, this is a manageable solution that allows people to at least move away from the static default of `US`.

File: packages/peregrine/lib/store/actions/checkout/__tests__/asyncActions.spec.js [hunk 1]
```diff
@@ -25,6 +25,8 @@ const dispatch = jest.fn();
 const getState = jest.fn();
 const thunkArgs = [dispatch, getState];
 
+const oldDefaultCountry = process.env.DEFAULT_COUNTRY_CODE;
```
[the remark this answers is not in the document]
Comment (HaidarZ) [reply packages/peregrine/lib/store/actions/checkout/__tests__/asyncActions.spec.js:28]: Exactly ! I preferred the way that you've suggested and that's why I referenced this method in the [issue](https://github.com/magento/pwa-studio/issues/3059#issuecomment-824765856) asking for advice.

File: packages/venia-concept/webpack.config.js [hunk 2]
```diff
@@ -79,7 +79,10 @@ module.exports = async env => {
                 ? JSON.stringify(process.env.STORE_VIEW_CODE)
                 : JSON.stringify(storeConfigData.code),
             AVAILABLE_STORE_VIEWS: JSON.stringify(availableStores),
-            DEFAULT_LOCALE: JSON.stringify(global.LOCALE)
+            DEFAULT_LOCALE: JSON.stringify(global.LOCALE),
+            DEFAULT_COUNTRY_CODE: JSON.stringify(
+                process.env.DEFAULT_COUNTRY_CODE || 'US'
+            )
```
Comment (supernova-at) [inline packages/venia-concept/webpack.config.js:85]: In the event we do not have a `process.env.DEFAULT_COUNTRY_CODE`, this line will attempt to `JSON.stringify('US')`, which results in `"\"US\""`.

Instead, we want to do this like `STORE_VIEW_CODE` above:

```
DEFAULT_COUNTRY_CODE: process.env.DEFAULT_COUNTRY_CODE
  ? JSON.stringify(process.env.DEFAULT_COUNTRY_CODE)
  : 'US'
```

File: packages/venia-ui/.storybook/webpack.config.js [hunk 3]
```diff
@@ -49,7 +49,10 @@ module.exports = async ({ config: storybookBaseConfig, mode }) => {
                 ? JSON.stringify(process.env.STORE_VIEW_CODE)
                 : JSON.stringify(storeConfigData.code),
             AVAILABLE_STORE_VIEWS: JSON.stringify(availableStores),
-            DEFAULT_LOCALE: JSON.stringify(global.LOCALE)
+            DEFAULT_LOCALE: JSON.stringify(global.LOCALE),
+            DEFAULT_COUNTRY_CODE: JSON.stringify(
+                process.env.DEFAULT_COUNTRY_CODE || 'US'
+            )
```
Comment (supernova-at) [inline packages/venia-ui/.storybook/webpack.config.js:55]: In the event we do not have a `process.env.DEFAULT_COUNTRY_CODE`, this line will attempt to `JSON.stringify('US')`, which results in `"\"US\""`.

Instead, we want to do this like `STORE_VIEW_CODE` above:

```
DEFAULT_COUNTRY_CODE: process.env.DEFAULT_COUNTRY_CODE
  ? JSON.stringify(process.env.DEFAULT_COUNTRY_CODE)
  : 'US'
```

**Review by supernova-at — CHANGES_REQUESTED** [review]
Thanks for your contribution! Just a quick little fix and we're 👍

File: packages/venia-concept/webpack.config.js (hunk 2 above)
Comment (HaidarZ) [reply packages/venia-concept/webpack.config.js:85]: Regarding `STORE_VIEW_CODE` in both cases in the ternary operator it's applying `JSON.stringify`.  
```
...
STORE_VIEW_CODE: process.env.STORE_VIEW_CODE
    ? JSON.stringify(process.env.STORE_VIEW_CODE)
    : JSON.stringify(storeConfigData.code),
...
```
Shall I ignore this implementation and apply the code changes that you've requested ?

Comment (supernova-at) [reply packages/venia-concept/webpack.config.js:85]: Oops, nevermind, this is a quirk of the webpack DefinePlugin. These values should work 👍 .

**Review by supernova-at — DISMISSED** [review]

**Review by sirugh — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved

**Pull request merged**